### PR TITLE
Feature add mailer

### PR DESCRIPTION
**Pull request by fastndcurious**

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -63,5 +63,11 @@
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
+  # Mailer
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+  config.action_mailer.delivery_method = :smtp
+    config.action_mailer.smtp_settings = {
+        :address => "localhost",
+        :port => 1025
+    }
```
Comment (Crysicia) [inline config/environments/development.rb:72]: Prefer the new hash syntax

File: config/environments/test.rb [hunk 2]
```diff
@@ -43,4 +43,12 @@
...
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  # Mailer
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+  config.action_mailer.delivery_method = :smtp
+    config.action_mailer.smtp_settings = {
+        :address => "localhost",
+        :port => 1025
+    }
```
Comment (Crysicia) [inline config/environments/test.rb:53]: Prefer the new hash syntax

File: app/models/user.rb [hunk 3]
```diff
@@ -26,4 +26,9 @@ class User < ApplicationRecord
   has_one :wishlist, through: :profile
   has_many :carts, dependent: :destroy
   has_many :items, through: :carts
+  include Notifications
+
+  def offer(text)
+    UserMailer.offer(self, text).deliver_later
+  end
```
Comment (Crysicia) [inline app/models/user.rb:35]: ??? Where is this mailer ?

**Review by Crysicia — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (stephan-dev) [reply app/models/user.rb:35]: I concur

File: config/environments/development.rb (hunk 1 above)
Comment (azuryue) [reply config/environments/development.rb:72]: ok done

File: app/models/user.rb (hunk 3 above)
Comment (azuryue) [reply app/models/user.rb:35]: oops, added

**Review by stephan-dev — APPROVED** [review]
approved with the changes

**Review by Crysicia — DISMISSED** [review]

**Comment by Crysicia** [comment]
@CitizenYY I think you forgot to add SendGrid ;)

**Pull request merged**